ospatch/exec_step.go

```diff
@@ -0,0 +1,79 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package ospatch
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+	osconfigpb "github.com/GoogleCloudPlatform/osconfig/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/v1alpha2"
+	"github.com/GoogleCloudPlatform/osconfig/common"
+)
+
+func getExecutablePath(ctx context.Context, stepConfig *osconfigpb.ExecStepConfig) (string, error) {
+	if gcsObject := stepConfig.GetGcsObject(); gcsObject != nil {
+		var reader io.ReadCloser
+		reader, err := common.FetchWithGCS(ctx, gcsObject.GetBucket(), gcsObject.GetObject(), gcsObject.GetGenerationNumber())
+		if err != nil {
+			return "", fmt.Errorf("error reading GCS object: %s", err)
+		}
+		defer reader.Close()
+		logger.Debugf("Fetched GCS object bucket %s object %s generation number %d", gcsObject.GetBucket(), gcsObject.GetObject(), gcsObject.GetGenerationNumber())
```

Comment:
Patch jobs have their own logger (in patchRun, adds appropriate labels), I'm ok with leaving this as is for now as the logging logic is a bit odd and I do want to make it cleaner (maybe instead of patch job id exec id?), but add a TODO here to log this in the patch job/workflow step context.

In other code we pass the logger in if not using a patchRun method:
https://github.com/GoogleCloudPlatform/osconfig/blob/master/ospatch/patch_run.go#L414

Any suggestions to make this cleaner/more configurable will be appreciated :)

Reply:
After talking with Paul, I think I have an idea on how to make logging simpler. This will probably change with the move to the agent endpoint so feel free to leave as is for now.

Reply:
Sounds good, I'll leave as is and keep an eye out for the refactoring